### PR TITLE
statistics: estimate by lower bound.

### PR DESCRIPTION
**Pull request by hanfei1991**
If a point hits the gap between two buckects, it returns 0.

### CODE REVIEW

File: statistics/statistics_test.go [hunk 1]
```diff
@@ -130,7 +130,7 @@ func (s *testStatisticsSuite) TestBuild(c *C) {
 	c.Check(len(col.Buckets), Equals, 232)
 	count, err := col.equalRowCount(sc, types.NewIntDatum(1000))
 	c.Check(err, IsNil)
-	c.Check(int(count), Equals, 1)
+	c.Check(int(count), Equals, 0)
```
Comment (shenli) [inline statistics/statistics_test.go]: lessRowCount is not covered.

**Comment by hanfei1991** [comment]
@shenli PTAL

**Comment by shenli** [comment]
LGTM
@XuHuaiyu @zimulala PTAL

**Comment by zimulala** [comment]
LGTM

**Pull request merged**